test/e2e/install/install_test.go

```diff
@@ -188,22 +194,78 @@ var _ = Describe("[Install] durability", func() {
 			}
 		})
 
-		It("should be able to delete NUMAResourceOperator CR and redeploy without polluting cluster state", func() {
-			var nname client.ObjectKey
-			for _, obj := range deployedObj {
-				if nroObj, ok := obj.(*nropv1alpha1.NUMAResourcesOperator); ok {
-					nname = client.ObjectKeyFromObject(nroObj)
+		It("should restart RTE DaemonSet when image is updated in NUMAResourcesOperator", func() {
+
+			By("wait for DaemonSet to be ready")
+			nname := getNRONamespacedNameFromDeployedObjects(deployedObj)
+			Expect(nname.Name).NotTo(BeEmpty())
+
+			nroObj := &nropv1alpha1.NUMAResourcesOperator{}
+			err := e2eclient.Client.Get(context.TODO(), nname, nroObj)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("waiting for the DaemonSet to be created..")
+			uid := nroObj.GetUID()
+			ds := &appsv1.DaemonSet{}
+
+			dsCreatedTimeOut := 5 * time.Minute
+			dsCreatedPollPeriod := 10 * time.Second
+			Eventually(func() error {
+				var err error
+				ds, err = getDaemonSetByOwnerReference(uid)
+				return err
+			}, dsCreatedTimeOut, dsCreatedPollPeriod).Should(BeNil())
+
+			dsReadyTimeOut := 5 * time.Minute
+			dsReadyPollPeriod := 10 * time.Second
+			By("... and ready")
```

Comment:
nice, but you can probably coalesce the two waiting loops. We don't care too much about created but undready DS